# Thenus_australiensis_sex_marker

## 1. Quailty control of illumina reads with fastp (v 0.23.1)
```bash
fastp \
        -i \"$i1\" -I \"$i2\" \
        -o result/fastp/trim.${sample}_1.fq.gz \
        -O result/fastp/trim.${sample}_2.fq.gz \
        --trim_front1 8 --trim_front2 8 \
        --length_required 50 \
        --qualified_quality_phred 25 \
        --thread 16 \
        --html result/multiqc/${sample}.report.html \
        --json result/multiqc/${sample}.report.json" >> fastp.txt
```
## 2. Genome survey using kmer distribution generated by the Illumina reads with SOAPec (v 2.01) and GenomeScope (v 2.0)
```bash
jellyfish count \
      -m $i \
      -s 10G \
      -t 32 \
      -C \
      -o "$JF_FILE" \
      "${DATA_DIR}/merge_${n}_R1.fastq" \
      "${DATA_DIR}/merge_${n}_R2.fastq"

    jellyfish histo -t 64 "$JF_FILE" > "$HISTO_FILE"

    Rscript "genomescope2.0/genomescope.R"" \
      -i "$HISTO_FILE" \
      -o "$OUTPUT_DIR" \
      -k $i
```
## 3. Genome assembly
### 3.1 Genome assembly using abyss2
```bash
bbnorm.sh \
  in=merge_M4_R2.fastq.gz \
  out=/norm.merge_M4_R2.fastq.gz \
  target=40 mindepth=2 threads=24 -Xmx80G \
  hist=coverage.hist \
  prefilter=t passes=2

# Define input files
READS_R1="norm.merge_M4_R1.fastq.gz"
READS_R2="norm.merge_M4_R2.fastq.gz"
OUTPUT_DIR="abyss_results_65_M"
K_MER=65
# Use the number of CPUs requested

echo "Creating output directory: $OUTPUT_DIR"
mkdir -p $OUTPUT_DIR
cd $OUTPUT_DIR

echo "Starting ABySS assembly with k=$K_MER and $NUM_PROCESSORS threads..."
echo "Input reads: $READS_R1, $READS_R2"

# Run abyss-pe
# -C . is redundant if you cd into the directory, but doesn't hurt.
# Removed H=3 to let ABySS determine optimal hash functions for B=30G
# Added v=-v for verbose output including Bloom filter stats
 abyss-pe \
  name=Male_65\
  k=$K_MER \
  in="$READS_R1 $READS_R2" \
  B=50G \
  kc=3 \
  v=-v \
  j=43 np=2
```
### 3.2 Transcriptome assembly
```bash
nextflow run nf-core/denovotranscript \
-r 1.2.1 \
--input samplesheet.csv \
--outdir $nf \
-profile singularity \
--assemblers trinity \
--ss fr \
--busco_mode transcriptome \
--busco_lineage arthropoda_odb10 \
-c run.config \
```

```bash
cd-hit-est -i nf/results/ -o transcripts.cdhit97.fa \
  -c 0.97 -aS 0.95 -G 0 -g 1 -T 64 -M 0

salmon index -t transcripts.cdhit97.fa -i salmon/idx
salmon quant -i salmon/idx -l A \
-1 nf/results/cat/pooled_reads_1.merged.fastq.gz -2 nf/results/cat/pooled_reads_2.merged.fastq.gz \
-p 64 -o salmon

perl $TRINITY_HOME/util/abundance_estimates_to_matrix.pl \
  --est_method salmon \
  --name_sample_by_basedir \
  --quant_files salmon/quant_files.txt \
  --out_prefix matrix \
  --gene_trans_map none

# Keep transcripts with TPM â‰¥ 1 in at least one sample
perl $TRINITY_HOME/util/filter_low_expr_transcripts.pl \
  --matrix /mnt/12T/lobster_project/script_new/matrix.isoform.TMM.EXPR.matrix --transcripts /mnt/12T/lobster_project/script_new/busco/transcripts.cdhit97.fa --min_expr_any 1 \
  > salmon/okay.TPM1.fa

# Use a *stricter* identity (0.995) but keep 95% coverage of the shorter
cd-hit-est -i salmon/okay.TPM1.fa -o salmon/2okay.minredund.fa \
  -c 0.97 -aS 0.95 -G 0 -g 1 -T 64 -M 0
``` 

### 3.3 Polishing of assembly results using short-read RNA-seq
```bash
```
### 3.4 Polishing of assembly results using contig RNA-seq
```bash
```
### 3.5 Polishing of assembly results using short-read DNA-seq
```bash
```
## 4.Gene structure prediction
### 4.1 Gene structure prediction using AUGUSTUS (v3.2.3)
```bash
```
### 4.2 Gene structure prediction using PASA (v3.2.3)
```bash
```
### 4.3 Gene structure prediction using EVM (v3.2.3)
```bash
```
### 4.4 Gene structure prediction using PASA-update (v3.2.3)
```bash
```

## 5. Functional annotation
## 5.1 8.1 Diamond Blast (v0.88.2) the whole NCBI NR database (Release 2021_9_29)
```bash
```
## 5.1 8.1 Diamond Blast (v0.88.2) the SWISS-PROT database (Release 2022_01)
```bash
```
## 5.1 8.1 Protein domain annotation by interproscan (v5.52-86.0) and pfam_scan.pl
```bash
```
## Genome assessment
## 6. Noncoding RNAs prediction

### 6.1 tRNA annotation using tRNAscan-SE (v1.4)
### 6.2 rRNA, miRNA and snRNA annotation using INFERNAL (v1.0) form Rfam (14.5)
### 6.1 sncRNA annotation using tRNAscan-SE (v1.4)


## 7. Finding candidate sex-specific and positive control marker
https://github.com/fengtong-bio/ssp2

```bash
```
### 7.1 Step one: K-mer dataset preparation
```bash
```

### 6.2 Step two: Sex-specific k-mers and reads extraction.
```bash
```

### 7.3 Step three: De novo assembly of sex-specific reads.
```bash
```

### 7.4 Step four: Check against the candidates.
```bash
```
